File: tmdb-core/src/main/java/dev/reuss/tmdb/common/image/CollectionImages.java
package dev.reuss.tmdb.common.image;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Images for a TMDB collection.
 *
 * @param id        collection id
 * @param backdrops backdrop images
 * @param posters   poster images
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record CollectionImages(
        int id,
        List<TmdbImage> backdrops,
        List<TmdbImage> posters
) {

    public CollectionImages {
        backdrops = TmdbCollections.list(backdrops);
        posters = TmdbCollections.list(posters);
    }

    public boolean isEmpty() {
        return backdrops.isEmpty() && posters.isEmpty();
    }

    public boolean hasImages() {
        return !isEmpty();
    }

    public boolean hasBackdrops() {
        return !backdrops.isEmpty();
    }

    public boolean hasPosters() {
        return !posters.isEmpty();
    }

    public int backdropCount() {
        return backdrops.size();
    }

    public int posterCount() {
        return posters.size();
    }

    public int imageCount() {
        return backdropCount() + posterCount();
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/image/CollectionImages.kt
package dev.reuss.tmdb.common.image

/**
 * Images for a TMDB collection.
 *
 * @property id collection id
 * @property backdrops backdrop images
 * @property posters poster images
 */
@JvmRecord
data class CollectionImages(
    val id: Int,
    val backdrops: List<TmdbImage> = emptyList(),
    val posters: List<TmdbImage> = emptyList(),
) {
    fun isEmpty(): Boolean = backdrops.isEmpty() && posters.isEmpty()

    fun hasImages(): Boolean = !isEmpty()

    fun hasBackdrops(): Boolean = backdrops.isNotEmpty()

    fun hasPosters(): Boolean = posters.isNotEmpty()

    fun backdropCount(): Int = backdrops.size

    fun posterCount(): Int = posters.size

    fun imageCount(): Int = backdrops.size + posters.size
}
